src/game.cpp

```diff
@@ -7853,12 +7854,12 @@ void game::print_terrain_info( const tripoint &lp, WINDOW *w_look, int column, i
         wprintz(w_look, ll.second, ll.first.c_str());
     }
 
-    std::string signage = m.get_signage( lp );
-    if (!signage.empty() && signage.size() < 36) {
-        mvwprintw(w_look, ++line, column, _("Sign: %s"), signage.c_str());
-    } else if (!signage.empty()) {
+    std::string signage = g->u.has_trait( trait_ILLITERATE ) ? _( "???" ) : m.get_signage( lp );
```

Comment:
Don't access `u` via the global `g` pointer, this code is already inside a `game` member function, use just `u`.